File: 3254-find-the-power-of-k-size-subarrays-i/3254-find-the-power-of-k-size-subarrays-i.java
class Solution {
    public int[] resultsArray(int[] nums, int k) {
    if(k==1)
    return nums;    
    int n = nums.length; 
    int ans[] = new int[n-k+1];
        
    int start = 0, prev = 0, end = 1;
    int p = 0;

while(start<=(n-k) && end<n)
{
  if(nums[end] == nums[prev]+1)
  {
   if((end-start+1)==k)
   {
    ans[p++] = nums[end];
    start++ ;
   } 
    prev = end;  
    end++; 
  }
  
else{
      while(start<end && p<ans.length)
      {
        ans[p++] = -1; 
        start++; 
      }
      end = start + 1;
      prev = start; 
    }

}


   return ans;   
}
}

File: 3254-find-the-power-of-k-size-subarrays-i/3254-find-the-power-of-k-size-subarrays-i.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        if(k==1) return nums;
        vector<int>ans;
        bool flag=true;
        int j=1;
        int i=0;
        while(j<nums.size()) {
            while(j<nums.size()) {
                if(nums[j]!=nums[j-1]+1) {
                        flag=false;
                        break;
                    }
                j++;
            }
            //count tells the size of the continuous array that satisfy given condition
            int count=j-i+1-k;
            
            //for indices=count we will push back the max element in ans array
            while(count>0) {
                ans.push_back(nums[i+k-1]);
                count--;
                i++;
            }

            //for the remaining indices till j we will push -1            
            while(i<j && i+k<=nums.size()) {
                ans.push_back(-1);
                i++;
            }
            flag=true;
            j++;

        }
        return ans;     
    }
};